File: app/src/main/java/com/ephirium/storyline/common/delegate/AdapterDelegate.java
package com.ephirium.storyline.common.delegate;

import android.view.ViewGroup;

import androidx.recyclerview.widget.RecyclerView.ViewHolder;

public interface AdapterDelegate {

    ViewHolder onCreateViewHolder(ViewGroup parent);

    void onBindViewHolder(ViewHolder holder, DelegateItem item);

    Boolean isOfViewType(DelegateItem item);

}

File: app/src/main/java/com/ephirium/storyline/common/delegate/AdapterDelegate.kt
package com.ephirium.storyline.common.delegate

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView.ViewHolder

interface AdapterDelegate {
    fun onCreateViewHolder(parent: ViewGroup): ViewHolder
    fun onBindViewHolder(holder: ViewHolder, item: DelegateItem)
    fun isOfViewType(item: DelegateItem): Boolean
}
